draft-jennings-perc-double-01.md

```diff
@@ -96,127 +101,176 @@ Terms:
-encryption version of them can be added to the IANA registry.
+This specification uses two cryptographic contexts: An inner ("end-to-end")
+context that is used by endpoints that originate and consume media to ensure
+the integrity of media end-to-end, and an outer ("hop-by-hop") context" that
+is used between endpoints and MDDs to ensure the integrity of media over a
+single hop and to enable an MDD to modify certain RTP header fields.  The
+RECOMMENDED cipher for the hop-by-hop and end-to-end context is AES-GCM.
+However, other combinations of SRTP ciphers that support the procedures in
+this document can be added to the IANA registry.
 
 The keys and salt for these contexts are generated with the following steps:
 
-* Generate key and salt values of twice the length required by the E2E and HBH
-  transforms
+* Generate key and salt values of twice the length required by the inner
+  (end-to-end) and outer (hop-by-hop) transforms
 
 * Assign the first part of each value to be the key and salt, respectively, for
-  the inner transform.
+  the inner (end-to-end) transform.
 
 * Assign the second part of each value to be the key and salt, respectively, for
-  the outer transform.
-
+  the outer (hop-by-hop) transform.
+
+As a special case, the outer cryptographic transform MAY be the NULL cipher
+(see {{RFC3711}}) if a secure transport, such as {{RFC6347}}, is used over a
+hop (i.e., between an endpoint and MDD or between two MDDs).  In that case, the
+key and salt values generated would be the length required only for the inner
+cryptographic transform, which MUST NOT be the NULL cipher.
+  
 Obviously, if the MDD is to be able to modify header fields but not decrypt the
 payload, then it must have cryptographic context for the outer transform, but
 not the inner transform.  This document does not define how the MDD should be
-provisioned with this information.
+provisioned with this information.  {Note: At this point do we want to say
+something like "One possible way to provide key material for the outer
+("hop-by-hop") transform is to use {{I-D.jones-perc-dtls-tunnel}}."}
+
 
 # Original Header Block
 
 Any SRTP packet processed following these procedures MAY contain an Original
 Header Block (OHB) extension.
 
-This RTP header extension contains the original values
-of any modified header fields, in the following form:
+This RTP header extension contains the original values of any modified header
+fields and MUST be placed after any already-existing RTP header extensions.
+Placement of the OHB after any original header extensions is important so that
+the receiving endpoint can properly authenticate the original packet and
+any included RTP header extension values, which it will do by restoring the
+modified RTP header field values by copying those from the OHB and then
+removing the OHB extension and any other RTP header extensions that appear
+after the OHB extension.
```

Comment:
might be worth mentioning that this makes it so we do not need to explicitly carry the value of the original "Ext Len"